crates/runtime/src/dataconnector/parameters/aws.rs

```diff
@@ -0,0 +1,242 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use aws_config::{BehaviorVersion, Region, SdkConfig};
+use aws_credential_types::Credentials;
+use snafu::prelude::*;
+use tonic::async_trait;
+
+use crate::parameters::{ParamLookup, Parameters};
+
+use super::ConnectorParams;
+
+// https://docs.aws.amazon.com/general/latest/gr/rande.html
+pub const AWS_REGIONS: [&str; 32] = [
+    "us-east-1",
+    "us-east-2",
+    "us-west-1",
+    "us-west-2",
+    "af-south-1",
+    "ap-east-1",
+    "ap-south-1",
+    "ap-south-2",
+    "ap-northeast-1",
+    "ap-northeast-2",
+    "ap-northeast-3",
+    "ap-southeast-1",
+    "ap-southeast-2",
+    "ap-southeast-3",
+    "ap-southeast-4",
+    "ap-southeast-5",
+    "ca-central-1",
+    "ca-west-1",
+    "eu-central-1",
+    "eu-central-2",
+    "eu-west-1",
+    "eu-west-2",
+    "eu-west-3",
+    "eu-south-1",
+    "eu-south-2",
+    "eu-north-1",
+    "sa-east-1",
+    "il-central-1",
+    "me-south-1",
+    "me-central-1",
+    "us-gov-east-1",
+    "us-gov-west-1",
+];
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Invalid endpoint: {endpoint}"))]
+    InvalidEndpoint { endpoint: String },
+
+    #[snafu(display("Insecure endpoint without allow_http: {endpoint}"))]
+    InsecureEndpointWithoutAllowHTTP { endpoint: String },
+
+    #[snafu(display("Invalid region: {region}"))]
+    InvalidRegion { region: String },
+
+    #[snafu(display("Invalid region corrected: {region}"))]
+    InvalidRegionCorrected { region: String },
+
+    #[snafu(display(
+        "Invalid auth parameter combination: {parameter} requires auth 'key', not {auth}"
+    ))]
+    InvalidAuthParameterCombination { parameter: String, auth: String },
+
+    #[snafu(display("Missing access key"))]
+    NoAccessKey,
+
+    #[snafu(display("Missing access secret"))]
+    NoAccessSecret,
+
+    #[snafu(display("Unsupported authentication method: {method}"))]
+    UnsupportedAuthenticationMethod { method: String },
+}
+
+#[async_trait]
+pub(crate) trait Validator {
```

Comment:
This trait definition feels like it should be in `crates/runtime/src/dataconnector/parameters.rs`?